### PR TITLE
Add retrying API mode

### PR DESCRIPTION
**Pull request by dbradf**

### CODE REVIEW

File: src/evergreen/api.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import logging
 import time
 
+try:
+    from json.decoder import JSONDecodeError
+except ImportError:
```
Comment (lydiastepanek) [inline src/evergreen/api.py]: Might be helpful to include a comment like here https://github.com/evergreen-ci/evergreen.py/pull/21/files#diff-605ef5f63c2af7e5816682aaac567219R16

File: src/evergreen/api.py [hunk 2]
```diff
@@ -548,3 +561,24 @@ def clear_caches(self):
...
+
+
+class RetryingEvergreenApi(EvergreenApi):
+    """An Evergreen Api that retries failed calls."""
+
+    def __init__(self, api_server=DEFAULT_API_SERVER, auth=None, timeout=None):
+        """Create an Evergreen Api object."""
+        super(RetryingEvergreenApi, self).__init__(api_server, auth, timeout)
+
+    @retry(retry=retry_if_exception_type(requests.exceptions.HTTPError),
```
Comment (lydiastepanek) [inline src/evergreen/api.py]: fancy!

File: src/evergreen/api.py (hunk 1 above)
Comment (dbradf) [reply src/evergreen/api.py]: done

File: src/evergreen/api.py (hunk 2 above)
Comment (dbradf) [reply src/evergreen/api.py]: This is using the [tenacity|https://tenacity.readthedocs.io/en/latest/] which we have started using. It is pretty cool.

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**